lib/storage/keyval/etcd.go

```diff
@@ -63,6 +64,8 @@ func NewETCD(cfg ETCDConfig) (*electingBackend, error) {
 		Backend: &backend{
 			Clock:    clock,
 			kvengine: engine,
+
+			cachedCompleteOperations: make(map[string]*storage.SiteOperation),
```

Comment:
I would also consider factoring this out into some other type and let backend handle the retrieval. `cachingBackend` for instance can add caching non-invasively

Reply:
I don't really see how adding another backend that acts as a shim other than caching one function (or maybe more in the future) makes this less invasive. 

I think it also creates a crossing of concerns, because GetOperations which is the frequent call, can't be passed through with caching. The logic of how to GetOperations within the cache would drive a backend change to list just the operation keys, so the caching layer can re-implement GetOperations. GetOperations now is duplicated and tightly coupled to the backend implementation, and presumably need to be kept in sync. 

So I don't see why just getting caching within the key value store is an issue here.